src/digital.cpp

```diff
@@ -0,0 +1,90 @@
+#include "digital.h"
+
+// assign digital input to the correct pins
+DigitalIn setCruiseControlPin(D5);
+DigitalIn resetCruiseControlPin(D6);
+DigitalIn cruisePowerModePin(D3);
+DigitalIn cruiseSpeedModePin(D4);
+DigitalIn speedPin(D1);
+DigitalIn motorPowerPin(D12);
+DigitalIn forwardAndReversePin(D8);
+DigitalIn forwardAndReverseControlPin(D7);
+DigitalIn ecoModePin(D11);
+
+// storing every digital data inside a struct to reduce the memory
+volatile struct Digital_Data digital_data;
+
+// read set cruise control input
+void readSetCruiseControl(){
+    digital_data.setCruiseControlReading = setCruiseControlPin.read();
+}
+
+// read reset cruise control input
+void readResetCruiseControl(){
+    digital_data.resetCruiseControlReading = resetCruiseControlPin.read();
+}
+
+// read cruise power mode input
+void readCruisePowerMode(){
+    digital_data.cruisePowerModeReading = cruisePowerModePin.read();
+}
+
+// read cruise speed mode input
+void readCruiseSpeedMode(){
+    digital_data.cruiseSpeedModeReading = cruiseSpeedModePin.read();
+}
+
+// read speed input
+void readSpeed(){
+    digital_data.speedReading = speedPin.read();
+}
+
+// read motor power input
+void readMotorPower(){
+    digital_data.motorPowerReading = motorPowerPin.read();
+}
+
+// read forward and reverse input
+void readForwardAndReverse(){
+    digital_data.forwardAndReverseReading = forwardAndReversePin.read();
+}
+
+// read control forward and reverse input
+void readControlForwardAndReverse(){
+    digital_data.ControlforwardAndReverseReading = forwardAndReverseControlPin.read();
+}
+
+// read eco mode input
+void readEcoMode(){
+    digital_data.ecoModeReading = ecoModePin.read();
+}
+
+// read all the digital inputs
+void readDigital(){
+    readSetCruiseControl();
+    readResetCruiseControl();
+    readCruisePowerMode();
+    readCruisePowerMode();
+    readCruiseSpeedMode();
+    readSpeed();
+    readMotorPower();
+    readForwardAndReverse();
+    readControlForwardAndReverse();
+    readEcoMode();
+}
+
+// initialize Ticker to setup a recurring interrupt to repeatedly call the function at a specified rate
+Ticker readDigitalDelay;
+
+
+void flip() {
+  readDigital();
+}
```

Comment:
This function is unnecessary, set up the Ticker to call readDigital directly instead